Mono.Debugging.Soft/SoftDebuggerAdaptor.cs

```diff
@@ -916,9 +916,15 @@ static bool IsAnonymousType (TypeMirror type)
 
 		protected override ValueReference GetMember (EvaluationContext ctx, object t, object co, string name)
 		{
+			return GetMember (ctx, null, t, co, name);
+		}
+
+		public override ValueReference GetMember (EvaluationContext ctx, IObjectSource objectSource, object t, object co, string name)
+		{
 			var type = t as TypeMirror;
+			var tupleNames = GetTupleElementNames (objectSource, ctx);
```

Comment:
You should at least check that the type is a tuple before calling GetTupleElementNames.